skymap_scanner/server/start_scan.py

```diff
@@ -549,7 +601,12 @@ def collect(self, pixreco: pixelreco.PixelReco) -> None:
             raise DuplicatePixelRecoException(
                 f"Pixel-reco has already been received: {pixreco.id_tuple}"
             )
-        self.pixreco_ids_received.append(pixreco.id_tuple)
+        if pixreco.id_tuple not in self.pixreco_ids_sent:
+            raise DuplicatePixelRecoException(
+                f"Pixel-reco received not in sent set, it is probably from an earlier iteration: {pixreco.id_tuple}"
+            )
+            
+        self.pixreco_ids_received.add(pixreco.id_tuple)
```

Comment:
I'm curious, were you seeing this scenario in a test run? I can see how it's possible since `MQClient` doesn't guarantee deliver-once delivery.

Also, see my comment below on `n_pixreco`/`pixreco_ids_sent`

Reply:
Yes, I was seeing the server receive nside=8 pixels during the nside=64 iteration. The line above would only catch it if it was already in the list of received pixels. For the first (8, x, y) occurrence it would be added to the list, then later would raise a Duplicate exception when checking the `nsides_dict`. However, the length of the list was still incremented by 1.

Reply:
I imagine this isn't happening _too_ often (thinking about performance). It's good we're accounting for it now.